fs-index/src/tests.rs

```diff
@@ -913,3 +913,84 @@ fn test_track_modification_with_collision() {
         assert_eq!(index.collisions().len(), 1, "{:?}", index);
     });
 }
+
+/// Test for calling `update_one()` on a file that was moved from the root
+/// directory to a subdirectory.
+///
+/// ## Test scenario:
+/// - Create a file within the temporary directory.
+/// - Build a resource index in the temporary directory.
+/// - Move the file to a subdirectory.
+/// - Call `update_one()` 2 times with the relative path of the moved file.
+/// - Assert that the index contains the expected number of entries with the
+///  correct IDs and paths after the move.
+#[test]
+fn test_track_move_to_subdirectory() {
```

Comment:
Is this test-case really different from `fn test_track_rename()`?

Could you create test-case for modification? Btw I'm not sure that 2 calls are needed for modification, because path doesn't change. Let's double-check and correct the comment if needed.

Reply:
> Is this test-case really different from `fn test_track_rename()`?

It’s not really necessary, but when I was working on this initially, those were the cases where I ran into errors and had to debug for a bit 😃. So, I decided to explicitly include them in the tests. But yes, you’re right—they are essentially the same thing. Let’s just keep the `test_track_move_to_subdirectory` test.

> Could you create test-case for modification?

We already have `test_track_modification` and `test_track_modification_with_collision`. Do you think we need another one?

> Btw I'm not sure that 2 calls are needed for modification, because path doesn't change.

Yes, only one call to `update_one` is needed.